src/physics/em/detail/RelativisticBremInteractor.i.hh

```diff
@@ -0,0 +1,124 @@
+    , allocate_(allocate)
+    , dxsec_(shared, particle, material, elcomp_id)
+{
+    CELER_EXPECT(particle.particle_id() == shared_.ids.electron
+                 || particle.particle_id() == shared_.ids.positron);
+    CELER_EXPECT(gamma_cutoff_.value() > 0);
+
+    // Valid energy region of the relativistic e-/e+ Bremsstrahlung model
+    CELER_EXPECT(inc_energy_ > shared_.low_energy_limit());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the production of photons using the G4eBremsstrahlungRelModel
+ * of Geant4 6.10.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction RelativisticBremInteractor::operator()(Engine& rng)
+{
+    // Min and max kinetic energy limits for sampling the secondary photon
+    Energy tmin = min(gamma_cutoff_, inc_energy_);
+    Energy tmax = min(shared_.high_energy_limit(), inc_energy_);
+    if (tmin >= tmax)
+    {
+        return Interaction::from_unchanged(inc_energy_, inc_direction_);
+    }
+
+    real_type density_corr = dxsec_.density_correction();
+    real_type xmin         = std::log(ipow<2>(tmin.value()) + density_corr);
+    real_type xrange = std::log(ipow<2>(tmax.value()) + density_corr) - xmin;
+
+    real_type gamma_energy{0};
+    real_type dsigma{0};
+
+    do
+    {
+        gamma_energy = std::sqrt(max(
+            0.0,
+            std::exp(xmin + generate_canonical(rng) * xrange) - density_corr));
+        dsigma       = dxsec_(gamma_energy);
+    } while (dsigma < dxsec_.maximum_value() * generate_canonical(rng));
+
+    // Allocate space for the brems photon
+    Secondary* secondaries = this->allocate_(1);
+    if (secondaries == nullptr)
+    {
+        // Failed to allocate space for the secondary
+        return Interaction::from_failure();
+    }
+
+    // Construct interaction for change to parent (incoming) particle
+    Interaction result;
+    result.action      = Action::scattered;
+    result.energy      = units::MevEnergy{inc_energy_.value() - gamma_energy};
+    result.secondaries = {secondaries, 1};
+    secondaries[0].particle_id = shared_.ids.gamma;
+    secondaries[0].energy      = units::MevEnergy{gamma_energy};
+
+    // angular distribution: G4ModifiedTsai
+
+    // Generate exiting gamma direction from isotropic azimuthal
+    // angle and TsaiUrbanDistribution for polar angle
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    //    TsaiUrbanDistribution sample_gamma_angle(secondaries[0].energy,
+    TsaiUrbanDistribution sample_gamma_angle(inc_energy_,
+                                             shared_.electron_mass);
+    real_type             cost = sample_gamma_angle(rng);
+    secondaries[0].direction
+        = rotate(from_spherical(cost, sample_phi(rng)), inc_direction_);
+
+    // Update parent particle direction
+    real_type inc_momentum = std::sqrt(
+        inc_energy_.value()
+        * (inc_energy_.value() + 2 * shared_.electron_mass.value()));
```

Comment:
Store `inc_momentum_ = particle.momentum()` as part of the class data maybe, since it's part of the `Particle` class? That way if we want to try adding an optimization to precalculate momentum, it'll be automatically saved here.

Reply:
Added inc_momentum_ data and removed the explicit calculation line.